kview/src/main/java/dev/ikm/komet/kview/events/ShowPatternPanelEvent.java

```diff
@@ -51,4 +51,27 @@ public class ShowPatternPanelEvent extends Evt {
     public ShowPatternPanelEvent(Object source, EvtType eventType) {
         super(source, eventType);
     }
+
+    /**
+     * Constructs a prototypical Event.
+     *
+     * @param source    the object on which the Event initially occurred
+     * @param eventType
+     * @param fieldOrders the total number of field orders in the current pattern.
+     * @throws IllegalArgumentException if source is null
+     */
+    public ShowPatternPanelEvent(Object source, EvtType eventType, int fieldOrders) {
+        super(source, eventType);
+        this.fieldOrders = fieldOrders;
+    }
+
+    private int fieldOrders;
+
+    public int getFieldOrders() {
+        return fieldOrders;
+    }
+
+    public void setFieldOrders(int fieldOrders) {
+        this.fieldOrders = fieldOrders;
+    }
```

Comment:
Can you remove the setter method? In event programming event's and their data are usually imutable or copies of data. When publishing events the code will use the constructor. The subscriber (listener) will call the `getFieldOrders()`. In other words, the subscriber would never change the value of the payload (aka field order).

Reply:
Removed setter method.